src/HDFFile.cxx

```diff
@@ -201,6 +201,81 @@ static void populate_blob(std::vector<DT> &blob, rapidjson::Value const *vals) {
   }
 }
 
+static void populate_string_pointers(std::vector<char const *> &ptrs, rapidjson::Value const *vals) {
+  if (vals->IsString()) {
+    ptrs.push_back(vals->GetString());
+  } else if (vals->IsArray()) {
+    std::vector<rapidjson::Value const *> as;
+    std::vector<size_t> ai;
+    std::vector<size_t> an;
+    as.push_back(vals);
+    ai.push_back(0);
+    an.push_back(vals->GetArray().Size());
+
+    while (not as.empty()) {
+      if (as.size() > 10) {
+        break;
+      }
+      if (ai.back() >= an.back()) {
+        as.pop_back();
+        ai.pop_back();
+        an.pop_back();
+        continue;
+      }
+      auto &v = as.back()->GetArray()[ai.back()];
+      if (v.IsArray()) {
+        ai.back()++;
+        as.push_back(&v);
+        ai.push_back(0);
+        size_t n = v.GetArray().Size();
+        an.push_back(n);
+      } else if (v.IsString()) {
+        ptrs.push_back(v.GetString());
+        ai.back()++;
+      }
+    }
+  }
+}
+
+static void populate_string_fixed_size(std::vector<char> &blob, hsize_t element_size, rapidjson::Value const *vals) {
+  size_t n_added = 0;
+  if (vals->IsString()) {
+    memcpy(blob.data() + n_added * element_size, vals->GetString(), vals->GetStringLength());
```

Comment:
`std::copy` is preferred over `memcpy`, it is safer (works with non-POD types) and interestingly can be optimised better by the compiler in some situations (because it can make use of the retained type information).
It would be something like this (untested!):
```
auto string_in_json = vals->GetString();
std::copy(string_in_json, string_in_json + vals->GetStringLength(),
          std::back_inserter(blob));
```

Reply:
The advantages you list do not apply in the special case here, though. The advantage I do see is that `std::copy` is more idiomatic, should that piece of code get reused in the future.